Tests/scripts/validate_index.py

```diff
@@ -0,0 +1,163 @@
+"""Run validation on the index.json file.
+
+Check index.json file inside the index.zip archive in the cloud.
+Validate no missing ids are found and that all packs have a positive price.
+Validate commit hash is in master's history.
+"""
+import argparse
+import logging
+import json
+import sys
+import os
+
+from Tests.Marketplace.marketplace_services import init_storage_client, GCPConfig, CONTENT_ROOT_PATH
+from Tests.Marketplace.upload_packs import download_and_extract_index, get_content_git_client
+from Tests.scripts.utils.log_util import install_logging
+from pprint import pformat
+
+INDEX_FILE_PATH = 'index.json'
+DEFAULT_PAGE_SIZE = 50
+
+
+def options_handler():
+    parser = argparse.ArgumentParser(description='Utility for instantiating integration instances')
+    parser.add_argument('-e', '--extract_path',
+                        help=f"Full path of folder to extract the {GCPConfig.INDEX_NAME}.zip to",
+                        required=True)
+    parser.add_argument('-pb', '--production_bucket_name', help="Production bucket name", required=True)
+    parser.add_argument('-sa', '--service_account', help="Path to gcloud service account", required=True)
+
+    options = parser.parse_args()
+    return options
+
+
+def log_message_if_statement(statement, error_message, success_message=None):
+    """Log error message if statement is false, Log success otherwise
+
+    Args:
+        statement: The boolean statement to check.
+        error_message: The error message to log if statement is false
+        success_message: The success message to log if statement is true
+
+    Returns: The statements boolean value.
+    """
+    if not statement:
+        logging.error(error_message)
+    elif success_message is not None:
+        logging.success(success_message)
+    return statement
+
+
+def check_index_data(index_data):
+    """Check index.json file inside the index.zip archive in the cloud.
+
+    Validate by running verify_pack on each pack.
+
+    Args:
+        index_data: Dictionary of the index.json contents.
+
+    Returns: True if all packs are valid, False otherwise.
+    """
+    logging.info("Found index data in index file. Checking...")
+    logging.debug(f"Index data is:\n {pformat(index_data)}")
+
+    packs_list_exists = log_message_if_statement(statement=(len(index_data.get("packs", [])) != 0),
+                                                 error_message="Found 0 packs in index file."
+                                                               "\nAborting the rest of the check.")
+    if not packs_list_exists:
+        return False
+
+    packs_are_valid = True
+    for pack in index_data["packs"]:
+        pack_is_good = verify_pack(pack)
+        if not pack_is_good:
+            packs_are_valid = False
```

Comment:
you can return here, no need to keep iterating the rest

Reply:
Don't we want the whole picture?

Reply:
right, missed the logs